docs-src/modules/developing/pages/queries.adoc

```diff
@@ -1,11 +1,155 @@
 = Queries
 
-The following sections will help with writing queries.
+Kaskada computes results from the data you've loaded by executing _queries_.
+A query describes a computation to perform and configures how the query should be executed.
 
-== Final vs Historical Results
+Kaskada queries are witten in a language called Fenl, which is designed to make it easy to describe computations over events.
+To learn more about Fenl:
+
+* The xref:fenl:fenl-quick-start.adoc[Quick Start] is an overview of the query language.
+* The xref:fenl:language-guide.adoc[Language Guide] explains the design and theory behind the languge.
+* The xref:fenl:catalog.adoc[Function Catalog] documents the operations and functions provided by Fenl.
+* The xref:fenl:fenl-faq.adoc[FAQ] answers some comment questions about using Fenl.
+
+== Query syntax quickstart
+
+Kaskada's query language builds on the lessons of 50+ years of query language design to provide a declarative, composable, easy-to-read, and type-safe way of describing computations related to time. 
+The following is a quick overview of the query language's main features and syntax.
+
+=== Stateful aggregations
+
+Aggregate events to produce a continuous timeline whose value can be observed at arbitrary points in time.
+
+[source,Fenl]
+----
+{
+  max_review_to_date: max(review.stars),
+}
+----
+
+[TIP]
+.What are the curly brackets for?
+====
+Every query needs to return a xref:fenl:working-with-records.adoc[Record].
+Records allow one or more values to be grouped into a single row.
+You can create a record using the syntax `{key: value, key2: value2}`.
+====
+
+=== Automatic joins
+
+Every expression is associated with an “entity”, allowing tables and expressions to be automatically joined. Entities eliminate redundant boilerplate code.
+
+[source,Fenl]
+----
+{
+  purchases_per_page_view: count(purchase) / count(pageview),
+}
+----
+
+=== Event-based windowing
+
+Collect events as you move through time, and aggregate them with respect to other events. Ordered aggregation makes it easy to describe temporal interactions.
+
+[source,Fenl]
+----
+{
+  pageviews_since_last_purchase:  count(pageview, window=since(purchase)),
+  spend_since_last_review:        count(purchase, window=since(review)),
+}
+----
+
+=== Pipelined operations
+
+Pipe syntax allows multiple operations to be chained together. Write your operations in the same order you think about them. It's timelines all the way down, making it easy to aggregate the results of aggregations.
+
+[source,Fenl]
+----
+{
+  largest_spend_over_2_purchases: purchase.amount
+  | when(purchase.category == "food") 
+  | sum(window=sliding(2, $input)) # Inner aggregation
+  | max()                          # Outer aggregation
+}
+----
+
+=== Row generators
+
+Pivot from events to time-series. Unlike grouped aggregates, generators produce rows even when there's no input, allowing you to react when something doesn't happen.
```

Comment:
This is a first for me :) 
What is the generator? `mean()`? 
If we are to use the term "generators" we should point out which part of the expression is the generator. Maybe also a link to other supported generators?